test/e2e_node/cpu_manager_test.go

```diff
@@ -596,17 +597,30 @@ func runCfsQuotaGuPods(ctx context.Context, f *framework.Framework, disabledCPUQ
 	var err error
 	var ctnAttrs []ctnAttribute
 	var pod1, pod2, pod3 *v1.Pod
-	var cleanupPods []*v1.Pod
+	podsToClean := make(map[apitypes.UID]*v1.Pod) // pod.UID -> pod
+
+	cleanupTestPod := func(ctx2 context.Context, pod *v1.Pod) {
+		framework.Logf("deleting pod: %s/%s", pod.Namespace, pod.Name)
+		deletePodSyncByName(ctx2, f, pod.Name)
+		waitForContainerRemoval(ctx2, pod.Spec.Containers[0].Name, pod.Name, pod.Namespace)
+		framework.Logf("deleted pod: %s/%s", pod.Namespace, pod.Name)
+	}
+
+	deleteTestPod := func(pod *v1.Pod) {
+		// waitForContainerRemoval takes "long" to complete; if we use the parent ctx we get a
+		// 'deadline expired' message and the cleanup aborts, which we don't want.
```

Comment:
Same as above?

Reply:
this one wants to be a closure thought because it manages the `podsToClean` map. I think most of the value comes from the previous helper anyway.